src/exporters/vhi.py

```diff
@@ -0,0 +1,172 @@
+        # extract the week_number
+        week_num = year_pattern.split(filename)[-1].split('.')[0]
+        # extract the year from the filename
+        year = year_pattern.findall(filename)[0].split('.P')[-1]
+
+        return year, week_num
+
+    def make_filename(self, raw_filename: str, dataset: str = 'vhi',) -> Path:
+        # check that the string is a legitimate name
+        assert len(filename.split('/')) == 1, f"filename cannot have subdirectories in it '/'. Must be the root filename. Currently: {filename}"
+
+        # make the dataset folder ('VHI')
+        dataset_folder = self.raw_folder / dataset
+        if not dataset_folder.exists():
+            dataset_folder.mkdir()
+
+        # make the year folder
+        year = _parse_time_from_filename(filename)[0]
+        assert isinstance(year, str), f"year must be a string! currently: < {year}, {type(year)} >"
+        year_folder = dataset_folder / year
+        if not year_folder.exists():
+            year_folder.mkdir()
+
+        # make the filename e.g. 'raw/vhi/1981/VHP.G04.C07.NC.P1981035.VH.nc'
+        filename = year_folder / raw_filename
+
+        return filename
+
+    def download_file_from_ftp(ftp_instance: FTP,
+                               filename: str,
+                               output_filename: Path) -> None:
+        # check if already exists
+        if output_filename.exists():
+            print(f"File already exists! {output_filename}")
+            return
+
+        # download to output_filename
+        with open(output_filename,'wb') as out_f:
+            ftp_instance.retrbinary("RETR " + filename, out_f.write)
+
+        if output_filename.exists():
+            print(f"Successful Download! {output_filename}")
+        else:
+            print(f"Error Downloading file: {output_filename}")
+
+        return
+
+    def batch_ftp_request(filenames: List) -> None:
+        # create one FTP connection for each batch
+        with FTP('ftp.star.nesdis.noaa.gov') as ftp:
+            ftp.login()
+            ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+            # download each filename using this FTP object
+            for filename in filenames:
+                output_filename = (
+                    self.make_filename(filename, dataset='vhi')
+                )
+                download_file_from_ftp(ftp, filename, output_filename)
+
+        return
+
+    def save_errors(outputs: List) -> None:
+        print("\nError: ",[errors for errors in outputs if errors != None])
+
+        # save the filenames that failed to a pickle object
+        with open(self.raw_folder / 'vhi_export_errors.pkl','wb') as f:
+            pickle.dump([error[-1] for error in outputs if error != None], f)
+
+        return
+
+    def run_parallel(vhi_files: List) -> List:
+        pool = Pool(processes=100)
+
+        # split the filenames into batches of 100
+        batches = [batch for batch in chunks(vhi_files,100)]
```

Comment:
this is also working except for a type incompatbility.

```
src/exporters/vhi.py:172: error: Argument 1 to "chunks" of "VHIExporter" has incompatible type "List[Any]"; expected "int"
```

Reply:
Oh it's all fixed by changing to : 
`def chunks(self, l: List, n: int) -> Generator:`